src/flavors/list.rs

```diff
@@ -166,11 +177,23 @@ impl<T> Channel<T> {
 
             // A closure that installs a block following `tail` in case it hasn't been yet.
             let install_next_block = || {
+                let new_slot_count = {
+                    let double_slot_count = tail_slot_count << 1;
+                    if double_slot_count < BLOCK_CAP {
+                        double_slot_count
+                    } else {
+                        BLOCK_CAP
+                    }
+                };
```

Comment:
This can be just: `let new_slot_count = (tail_slot_count * 2).min(BLOCK_CAP)`

Reply:
That is much cleaner - thanks!